File: bephimfe/src/service/CategoryService.ts
import axios from "axios";
import type { CategoryRequest,CategoryResponse } from "../types/Category";
const apiAdmin=axios.create({
    baseURL:"http://localhost:8080/admin/category"
})
apiAdmin.interceptors.request.use(config => {
  const token = sessionStorage.getItem("token");
  if (token) {
    config.headers.Authorization = `Bearer ${token}`;
  }
  return config;
});
const CategoryService={
    getAll:async():Promise<CategoryResponse>=>{
        const response=await apiAdmin.get(``)
        return response.data
    }
}
export default CategoryService

File: bephim/src/main/java/org/example/bephim/service/admin/category/CategoryService.java
package org.example.bephim.service.admin.category;

import org.example.bephim.dto.category.request.CategoryRequest;
import org.example.bephim.dto.category.response.CategoryResponse;

import java.util.List;

public interface CategoryService {
     List<CategoryResponse> getAll();
     CategoryResponse getById(Integer id);
     CategoryResponse getByName(String name);
     CategoryResponse create(CategoryRequest categoryRequest);
     CategoryResponse update(Integer id,CategoryRequest categoryRequest);
}
